cloud-service-providers/google-cloud/vertexai/python/README.md

```diff
@@ -0,0 +1,194 @@
+
+## Prerequisites
+* [NGC API KEY](https://org.ngc.nvidia.com/setup/personal-keys)
+* [NGC CLI](https://org.ngc.nvidia.com/setup/installers/cli)
+* [Vertex AI Workbench](https://cloud.google.com/vertex-ai/docs/workbench/introduction)
+* [gcloud CLI](https://cloud.google.com/sdk/docs/install)
+
+## Run NIM on Vertex AI Workbench Instance
+### 1. Create an Vertex AI Workbench Instance
+Create an new Vertex AI Workbench instance and select `ADVANCED OPTIONS`. Choose NVIDIA GPUs (e.g. L4 for G2 machine series) and recommended [Disk Space](https://docs.nvidia.com/nim/large-language-models/latest/support-matrix.htmlß) for specific NIM.
+[<img src="imgs/vertexai_01.png" width="750"/>](HighLevelArch)
+
+### 2. Run NIM on JupyterLab Notebook
+`OPEN JUPYTERLAB` of the instance, and install required packages per `requirements.txt`. 
+
+Run `nim-vertexai.ipynb` Python jupyter notebook, which provides step-to-step guidance on how to deploy and inference the NIM container within notebook interface or via Vertex AI endpoint resource.
+
+If NIM container has been successfully launched, you will see below output in cell or deployment log:
+
+```shell
+===========================================
+== NVIDIA Inference Microservice LLM NIM ==
+===========================================
+
+NVIDIA Inference Microservice LLM NIM Version 1.0.0
+Model: nim/meta/llama3-8b-instruct
+
+Container image Copyright (c) 2016-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+
+This NIM container is governed by the NVIDIA AI Product Agreement here:
+https://www.nvidia.com/en-us/data-center/products/nvidia-ai-enterprise/eula/.
+A copy of this license can be found under /opt/nim/LICENSE.
+
+The use of this model is governed by the AI Foundation Models Community License
+here: https://docs.nvidia.com/ai-foundation-models-community-license.pdf.
+
+ADDITIONAL INFORMATION: Meta Llama 3 Community License, Built with Meta Llama 3. 
+A copy of the Llama 3 license can be found under /opt/nim/MODEL_LICENSE.
+```
+
+**Airgap NIM Container**
+
+Airgap NIM container has the advantages of having `NGC_API_KEY` inbuilt, thus not required when run within interface or via endpoint. It could also help to avoid exposing credentail on Vertex AI `Model Registry`.
+
+If needed, please refer to `/airgap` directory to update volume bind source file path in `docker-compose.yaml`, and backend profile in `airgap_model_manifest.yaml`. 
+
+Then follow insturctions in the notebook `Build Airgap NIM Container` to build new airgap docker image. 
+
+### 3. Inference in Online prediction
+After deploying NIM container to endpoint, check Vertex AI `Model Registry` and `Online prediction` for model/endpoint version details and event logs.
+
+[<img src="imgs/vertexai_02.png" width="750"/>](HighLevelArch)
+
+Make endpoint inference with OpenAI Python API or CLI. Streaming reponse is also supported.
```

Comment:
It is not clear what this line is indicating. Are we saying there is an optional Streaming on/off support for the OpenAI support through the Python API?

Reply:
Yes. Streaming on/off support for Python API as well as CLI (e.g. CURL), both solutions provided in the notebook.

Reply:
Will add on/off to make it more understandable.

Reply:
updated to make it more understandable